### PR TITLE
[dev-v2.6] Forwardport CSI 2.1.1 and add CSI 100.0.1+up2.1.1

### PR DESCRIPTION
**Pull request by PennyScissors**
Issue: https://github.com/rancher/rancher/issues/35378

Forwarports CSI 2.1.1 from 2.5, and adds CSI 100.0.1+up2.1.1 which is the 2.6 variant of the 2.5's CSI 2.1.1

### CODE REVIEW

File: charts/rancher-vsphere-csi/rancher-vsphere-csi/100.0.1+up2.1.1/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+annotations:
+  catalog.cattle.io/certified: rancher
+  catalog.cattle.io/display-name: vSphere CSI
+  catalog.cattle.io/namespace: kube-system
+  catalog.cattle.io/os: linux
+  catalog.cattle.io/rancher-version: '>= 2.6.0-0'
```
Comment (jiaqiluo) [inline charts/rancher-vsphere-csi/rancher-vsphere-csi/100.0.1+up2.1.1/Chart.yaml:6]:
```suggestion
  catalog.cattle.io/rancher-version: '>= 2.6.0-0 <= 2.6.99-0'
```

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

**Review by jiaqiluo — DISMISSED** [review]

**Review by aiyengar2 — CHANGES_REQUESTED** [review]
https://github.com/rancher/charts/blob/f0f7b6078ab19893009b654cd6c89e96a7970ec5/packages/rancher-vsphere-csi/package.yaml#L2 100.0.1 is already in dev.

We need to modify that first before introducing another 100.0.1 chart.

**Comment by PennyScissors** [comment]
> https://github.com/rancher/charts/blob/f0f7b6078ab19893009b654cd6c89e96a7970ec5/packages/rancher-vsphere-csi/package.yaml#L2
> 100.0.1 is already in dev.
> 
> We need to modify that first before introducing another 100.0.1 chart.

This should now be resolved

**Review by jiaqiluo — APPROVED** [review]

**Review by aiyengar2 — APPROVED** [review]
Ensuring the asset was copied verbatim from dev-v2.5...

```bash
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ git fetch upstream
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ git checkout upstream/dev-v2.5 -- assets/rancher-vsphere-csi/rancher-vsphere-csi-2.1.1.tgz charts/rancher-vsphere-csi/rancher-vsphere-csi/2.1.1
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ git status
HEAD detached at steven/devv26-forwardport-vsphere
nothing to commit, working tree clean
```

Ensuring the diff between the old asset and the new chart matches what we expect...

```bash
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ tar xvzf assets/rancher-vsphere-csi/rancher-vsphere-csi-2.1.1.tgz
x rancher-vsphere-csi/Chart.yaml
x rancher-vsphere-csi/values.yaml
x rancher-vsphere-csi/templates/_helpers.tpl
x rancher-vsphere-csi/templates/vsphere-csi-controller-deployment.yaml
x rancher-vsphere-csi/templates/vsphere-csi-controller-rbac.yaml
x rancher-vsphere-csi/templates/vsphere-csi-node-ds.yaml
x rancher-vsphere-csi/templates/vsphere-csi-secret.yaml
x rancher-vsphere-csi/templates/vsphere-csi-storageclass.yaml
x rancher-vsphere-csi/README.md
x rancher-vsphere-csi/app-readme.md
x rancher-vsphere-csi/questions.yaml
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ diff -uNr rancher-vsphere-csi charts/rancher-vsphere-csi/rancher-vsphere-csi/100.0.1+up2.1.1/
diff -uNr rancher-vsphere-csi/Chart.yaml charts/rancher-vsphere-csi/rancher-vsphere-csi/100.0.1+up2.1.1/Chart.yaml
--- rancher-vsphere-csi/Chart.yaml      2021-11-09 12:13:58.000000000 -0800
+++ charts/rancher-vsphere-csi/rancher-vsphere-csi/100.0.1+up2.1.1/Chart.yaml   2021-11-10 09:08:23.000000000 -0800
@@ -3,7 +3,7 @@
   catalog.cattle.io/display-name: vSphere CSI
   catalog.cattle.io/namespace: kube-system
   catalog.cattle.io/os: linux
-  catalog.cattle.io/rancher-version: '>= 2.5.0-0 <= 2.5.99-0'
+  catalog.cattle.io/rancher-version: '>= 2.6.0-0 <= 2.6.99-0'
   catalog.cattle.io/release-name: vsphere-csi
 apiVersion: v1
 appVersion: 2.1.1
@@ -18,4 +18,4 @@
 name: rancher-vsphere-csi
 sources:
 - https://github.com/kubernetes-sigs/vsphere-csi-driver
-version: 2.1.1
+version: 100.0.1+up2.1.1
```

Ensuring the new packaged asset matches the chart, which in turn matches the old asset aside from the expected diff (as checked in the previous command):

```bash
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ mkdir -p new
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ tar xvzf assets/rancher-vsphere-csi/rancher-vsphere-csi-100.0.1+up2.1.1.tgz  -C new 
x rancher-vsphere-csi/Chart.yaml
x rancher-vsphere-csi/values.yaml
x rancher-vsphere-csi/templates/_helpers.tpl
x rancher-vsphere-csi/templates/vsphere-csi-controller-deployment.yaml
x rancher-vsphere-csi/templates/vsphere-csi-controller-rbac.yaml
x rancher-vsphere-csi/templates/vsphere-csi-node-ds.yaml
x rancher-vsphere-csi/templates/vsphere-csi-secret.yaml
x rancher-vsphere-csi/templates/vsphere-csi-storageclass.yaml
x rancher-vsphere-csi/README.md
x rancher-vsphere-csi/app-readme.md
x rancher-vsphere-csi/questions.yaml
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ diff -uNr new/rancher-vsphere-csi charts/rancher-vsphere-csi/rancher-vsphere-csi/100.0.1+up2.1.1/
```

LGTM

**Pull request merged**